src/agentscope/service/web/tripadvisor.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+"""TripAdvisor APIs for searching and retrieving location information."""
+
+from typing import List, Any, Dict
+from loguru import logger
+import requests
+from agentscope.service.service_response import ServiceResponse
+from agentscope.service.service_status import ServiceExecStatus
+
+
+def get_tripadvisor_location_photos(
+    api_key: str,
+    location_id: str,
+    language: str = "en",
+) -> ServiceResponse:
+    """
+    Get photos for a specific location using
+    the TripAdvisor API and return the largest one.
+
+    Args:
+        api_key (`str`):
+            Your TripAdvisor API key.
+        location_id (`str`):
+            The location ID for the desired location.
+        language (`str`, optional):
+            The language for the response. Defaults to 'en'.
+
+    Returns:
+        `ServiceResponse`: A dictionary with two variables: `status` and
+        `content`. The `status` variable is from the ServiceExecStatus enum,
+        and `content` is a dictionary containing the largest photo information
+        or error information, which depends on the `status` variable.
+
+    Example:
+        .. code-block:: python
+
+            result = get_tripadvisor_location_photos(
+                "your_api_key", "12345", "en"
+            )
+            if result.status == ServiceExecStatus.SUCCESS:
+                print(result.content['largest_photo'])
+    """
+
+    def find_largest_photo(photos: List[Dict[str, Any]]) -> Dict[str, Any]:
```

Comment:
Any reason to why only return the largest photo?

Reply:
There is a chance the resolution of the small image is too low for some use cases. All the images from the pool are identical so the largest image preserves the most information.